tests/test_pipeline.py

```diff
@@ -150,6 +152,20 @@ def test_non_existing(self):
         ret = main(["pipeline", "show", "non-existing"])
         self.assertNotEqual(ret, 0)
 
+    def test_print_locked_stages(self):
+        self._caplog.clear()
+
+        self.dvc.add("foo")
+        self.dvc.add("bar")
+        self.dvc.lock_stage("foo.dvc")
+
+        with self._caplog.at_level(logging.INFO, logger="dvc"):
+            ret = main(["pipeline", "show", "foo.dvc", "--locked"])
+            self.assertEqual(ret, 0)
+
+        expected_stdout = "foo.dvc"
+        assert expected_stdout in self._caplog.text
```

Comment:
And what if `pipeline show` is going to print
```
foo.dvc
bar.dvc
```
? Your test would still pass, right?

Reply:
Yes. Wanted to a more strict comparison but followed the convention. 

Looking into how we can do a more strict comparison of command outputs. Both `"\n".join(record.message for record in self._caplog.records)` and `self._caplog.text` return the entire captures, which verbose and pretty much defeats the purpose of testing for outputs.

Reply:
@rpip Maybe something like:
```
assert "foo.dvc" in self._caplog.text
assert "bar.dvc" not in self._caplog.text
```
?

Reply:
that won't work; debug logs from [state.py#L164](https://github.com/iterative/dvc/blob/master/dvc/state.py#L146) contain "bar.dvc", and that shows in the captured log outputs.

```
self = <tests.test_pipeline.TestPipelineShow testMethod=test_print_locked_stages>

    def test_print_locked_stages(self):
        self._caplog.clear()

        self.dvc.add("foo")
        self.dvc.add("bar")
        self.dvc.lock_stage("foo.dvc")

        with self._caplog.at_level(logging.INFO, logger="dvc"):
            ret = main(["pipeline", "show", "foo.dvc", "--locked"])
            self.assertEqual(ret, 0)

        assert "foo.dvc" in self._caplog.text
>       assert "bar.dvc" not in self._caplog.text
E       AssertionError: assert 'bar.dvc' not in 'state.py             ... 37 INFO     foo.dvc\n'
E         'bar.dvc' is contained here:
E           state.py                   141 DEBUG    PRAGMA user_version;
E           state.py                   146 DEBUG    fetched: [(3,)]
E           state.py                   141 DEBUG    CREATE TABLE IF NOT EXISTS state (inode INTEGER PRIMARY KEY, mtime TEXT NOT NULL, size TEXT NO
T NULL, md5 TEXT NOT NULL, timestamp TEXT NOT NULL)
E           state.py                   141 DEBUG    CREATE TABLE IF NOT EXISTS state_info (count INTEGER)
E           state.py                   141 DEBUG    CREATE TABLE IF NOT EXISTS link_state (path TEXT PRIMARY KEY, inode INTEGER NOT NULL, mtime TE
XT NOT NULL)
E           state.py                   141 DEBUG    INSERT OR IGNORE INTO state_info (count) SELECT 0 WHERE NOT EXISTS (SELECT * FROM state_info)
E           state.py                   141 DEBUG    PRAGMA user_version = 3;
E           local.py                   780 DEBUG    Skipping copying for '/private/var/folders/gk/m8b9nwgj3n37ssb4mbn5h6dr0000gn/T/dvc-test.62418.
0fqdk2jh.a80c41ca-e87a-4008-9ba4-fda5b0c952d8/foo', since it is not a symlink or a hardlink.
E           fs.py                       19 DEBUG    Path /private/var/folders/gk/m8b9nwgj3n37ssb4mbn5h6dr0000gn/T/dvc-test.62418.0fqdk2jh.a80c41ca
-e87a-4008-9ba4-fda5b0c952d8/.dvc/cache/ac/bd18db4cc2f85cedef654fccc4a4d8 inode 8645937009
E           state.py                   141 DEBUG    SELECT mtime, size, md5, timestamp from state WHERE inode=8645937009
E           state.py                   146 DEBUG    fetched: [('1555416709490084864', '3', 'acbd18db4cc2f85cedef654fccc4a4d8', '155541671028488089
/Users/yao/dev/dvc: 0 4 1           1:ghc# 2:python3.6# 3:python2.7- 4:[tmux]* 5:bash# 6:bash# 7:bash# 8:bash# 9:bash#              | 16 Apr 12:12
```

Reply:
tried debugging this and pretty printing the caplog records:

```
       ...
        import dvc.logger
        formatter = dvc.logger.ColorFormatter()
        print(formatter.format(caplog.records[0]))
        print(formatter.format(caplog.records[1]))
        print(formatter.format(caplog.records[...]))
       ....
```

nothing useful we can use for strict comparison atm. or am I missing something?

Reply:
@rpip You just need to run `self._caplog.clear()` before `with` to cleanup old messages.

Reply:
ah yes, thanks. I called `self._caplog.clear()` too early. tried now, works -- pushing in a bit

Reply:
@efiop fixed.